plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerUtils.java

```diff
@@ -67,57 +72,44 @@
 	public static final String OPENSHIFT_SERVER_TYPE = "org.jboss.tools.openshift.openshift.server.type";//$NON-NLS-1$
 
 	public static final String SERVER_START_ON_CREATION = "org.jboss.tools.openshift.SERVER_START_ON_CREATION";
+
+	public static IServer findServerForService(String serviceName) {
+		final IServerType serverType = getServerType();
+		final Optional<IServer> match = Stream.of(ServerCore.getServers())
+				.filter(server -> server.getServerType()
+						.equals(serverType)
+						&& server.getAttribute(OpenShiftServerUtils.ATTR_SERVICE, "").equals(serviceName))
+				.findAny();
+		if (match.isPresent()) {
+			return match.get();
+		} else {
+			return null;
+		}
+	}
+	
+	public static IServerType getServerType() {
+		return ServerCore.findServerType(OpenShiftServer.SERVER_TYPE_ID);
+	}
 	
 	public static String getServerName(IService service, Connection connection) {
 		if (service == null) {
 			return null;
 		}
 
-		return new StringBuilder(service.getName())
+		String baseName = new StringBuilder(service.getName())
 				.append(" at OpenShift 3 (")
 				.append(UrlUtils.cutPort(UrlUtils.cutScheme(connection.getHost())))
```

Comment:
can these method return null or fail with NPE if the argument is null ? (I can't see these methods in my workspace yet)